components/integrations/bots/bot.tsx

```diff
@@ -31,62 +34,73 @@ export function matchesFilter(bot, filter, owner) {
         ownerUsername.toLowerCase().indexOf(filter) === -1);
 }
 
-export default class Bot extends React.PureComponent {
-    static propTypes = {
+type Props = {
 
-        /**
-        *  Bot that we are displaying
-        */
-        bot: PropTypes.object.isRequired,
+    /**
+    *  Bot that we are displaying
+    */
+    bot: BotType;
 
-        /**
-        * Owner of the bot we are displaying
-        */
-        owner: PropTypes.object,
+    /**
+    * Owner of the bot we are displaying
+    */
+    owner?: UserProfile;
 
-        /**
-        * User of the bot we are displaying
-        */
-        user: PropTypes.object,
+    /**
+    * User of the bot we are displaying
+    */
+    user: UserProfile;
+
+    /**
+    * The access tokens of the bot user
+    */
+    accessTokens: Dictionary<UserAccessToken>;
+
+    /**
+    * String used for filtering bot items
+    */
+    filter?: string;
+
+    actions: {
 
         /**
-        * The access tokens of the bot user
+        * Disable a bot
         */
-        accessTokens: PropTypes.object.isRequired,
+        disableBot: (userId: string) => Promise<ActionResult>;
 
         /**
-        * String used for filtering bot items
+        * Enable a bot
         */
-        filter: PropTypes.string,
-
-        actions: PropTypes.shape({
-
-            /**
-            * Disable a bot
-            */
-            disableBot: PropTypes.func.isRequired,
-
-            /**
-            * Enable a bot
-            */
-            enableBot: PropTypes.func.isRequired,
-
-            /**
-            * Access token managment
-            */
-            createUserAccessToken: PropTypes.func.isRequired,
-            revokeUserAccessToken: PropTypes.func.isRequired,
-            enableUserAccessToken: PropTypes.func.isRequired,
-            disableUserAccessToken: PropTypes.func.isRequired,
-        }),
+        enableBot: (userId: string) => Promise<ActionResult>;
 
         /**
-        *  Only used for routing since backstage is team based.
+        * Access token managment
         */
-        team: PropTypes.object.isRequired,
+        createUserAccessToken: (userId: string, description: string) => Promise<{
+            data: {token: string; description: string; id: string; is_active: boolean} | null;
+            error?: Error;
+        }>;
+
+        revokeUserAccessToken: (tokenId: string) => Promise<{data: string; error?: Error}>;
+        enableUserAccessToken: (tokenId: string) => Promise<{data: string; error?: Error}>;
+        disableUserAccessToken: (tokenId: string) => Promise<{data: string; error?: Error}>;
     };
 
-    constructor(props) {
+    /**
+    *  Only used for routing since backstage is team based.
+    */
+    team: Team;
+}
+
+type State = {
+    confirmingId: string;
+    creatingTokenState: string;
+    token: UserAccessToken | Record<string, any>;
+    error: string | JSX.Element;
```

Comment:
You can use `React.ReactNode` here to represent anything that can be rendered to the DOM.